Calling Player#applyMending(int) with zero exp causes ArithmeticException: / by zero

### Expected behavior

The applyMending method should not throw any exceptions when passing 0 as a value. At least that how it was in 1.20.4, since I just started running into that issue since updating to 1.21.4, and haven't changed anything to the plugin code.

I mean it would be a simple fix to not call the method when the value is 0, and I should implement that nonetheless, but throwing that exception seems wrong.

### Observed/Actual behavior

```
Caused by: java.lang.ArithmeticException: / by zero
	at org.bukkit.craftbukkit.entity.CraftPlayer.applyMending(CraftPlayer.java:1899) ~[paper-1.21.4.jar:1.21.4-31-d0d0efe]
	at erenos-3.6.0-SNAPSHOT.jar/net.aerulion.erenos.command.commands.DebugSubCommand.execute(DebugSubCommand.java:37) ~[erenos-3.6.0-SNAPSHOT.jar:?]
	at erenos-3.6.0-SNAPSHOT.jar/net.aerulion.erenos.command.ErenosSubCommand.validateAndExecute(ErenosSubCommand.java:63) ~[erenos-3.6.0-SNAPSHOT.jar:?]
	at erenos-3.6.0-SNAPSHOT.jar/net.aerulion.erenos.command.CommandService.onCommand(CommandService.java:249) ~[erenos-3.6.0-SNAPSHOT.jar:?]
	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[paper-api-1.21.4-R0.1-SNAPSHOT.jar:?]
	... 27 more
```

### Steps/models to reproduce

1. Get and hold a damaged pickaxe with mending: `/give @s minecraft:netherite_pickaxe[enchantments={levels: {"minecraft:mending": 1}},minecraft:damage=1]`
2. Call `Player#applyMending(int)` with 0 xp

### Plugin and Datapack List

Just a test plugin

### Paper version

[02:08:47 INFO]: This server is running Paper version 1.21.4-31-main@d0d0efe (2024-12-23T23:53:38Z) (Implementing API version 1.21.4-R0.1-SNAPSHOT)
You are running the latest version
Previous version: 1.21.4-30-aa2c52b (MC: 1.21.4)

### Other

_No response_

--- a/paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1896,7 +1896,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
                 handle.serverLevel(), itemstack, amount
             );
             int i = Math.min(possibleDurabilityFromXp, itemstack.getDamageValue());
-            final int consumedExperience = possibleDurabilityFromXp > 0 ? i * amount / possibleDurabilityFromXp : possibleDurabilityFromXp; // Paper - taken from ExperienceOrb#repairPlayerItems + prevent division by 0
+            final int consumedExperience = i > 0 ? i * amount / possibleDurabilityFromXp : possibleDurabilityFromXp; // Paper - taken from ExperienceOrb#repairPlayerItems + prevent division by 0
             org.bukkit.event.player.PlayerItemMendEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callPlayerItemMendEvent(handle, orb, itemstack, stackEntry.get().inSlot(), i, consumedExperience);
             i = event.getRepairAmount();
             orb.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.DESPAWN);

--- a/paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1896,7 +1896,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
                 handle.serverLevel(), itemstack, amount
             );
             int i = Math.min(possibleDurabilityFromXp, itemstack.getDamageValue());
-            final int consumedExperience = i * amount / possibleDurabilityFromXp; // Paper - taken from ExperienceOrb#repairPlayerItems
+            final int consumedExperience = possibleDurabilityFromXp > 0 ? i * amount / possibleDurabilityFromXp : possibleDurabilityFromXp; // Paper - taken from ExperienceOrb#repairPlayerItems + prevent division by 0
             org.bukkit.event.player.PlayerItemMendEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callPlayerItemMendEvent(handle, orb, itemstack, stackEntry.get().inSlot(), i, consumedExperience);
             i = event.getRepairAmount();
             orb.discard(org.bukkit.event.entity.EntityRemoveEvent.Cause.DESPAWN);

